### ISSUE TITLE
Grails 3.1.0.M3 plugin fails upon access when run as an app with 'run-app'

### ISSUE THREAD

**Issue by hakanernam** [grails/grails-core#9466]
A newly created plugin can not be run via 'run-app'. Message indicates successful start with:

```
Grails application running at http://localhost:8080 in environment: development
```

But attempting to access the app fails with display:

```
    - An error has occurred
    - Exception:
    - Message: No message available
     - Path: /
```

There are no messages printed on the console, in the meantime.

![screenshot at dec 07 23-47-27](https://cloud.githubusercontent.com/assets/1321157/11640753/ebdf27f6-9d3c-11e5-83e3-62070bd50278.png)

**Comment by graemerocher** [comment]
I guess this is because there is no UrlMappings.groovy

**Comment by hakanernam** [comment]
Yes. I copied from a new application and error disappeared. 

But assets directories are empty as well, so the view is not right. I thought I would add files to assets directories under

https://github.com/grails/grails-profile-repository/tree/master/profiles/plugin/skeleton/grails-app

But assets directory is not there, even when I have grails-app/assets directory and empty subdirectories in the generated plugin. Then I conclude I do not know what I am doing, and give up.

**Comment by ctoestreich** [comment]
Not all plugins are runnable.  Did you use the `grails create-plugin foo --profile=web-plugin`?  I think the secret sauce here is the correct profile.

**Comment by hakanernam** [comment]
@ctoestreich I just now, did exactly what you said. What happens is that the index page is shown but the assets "application.css" and "application.js" which is included by the layout main do not exist in assets directory, so the view is not rendered as intended:

![screenshot at dec 29 16-54-39](https://cloud.githubusercontent.com/assets/1321157/12036319/7bf157b2-ae4d-11e5-94f1-291020bb6724.png)

**Comment by hakanernam** [comment]
In any case, we are used to running plugins as applications. If we need to use special profile from now on, so that we can do what we have been doing all along, I believe it would need to be stressed in the documentation. I would be happy to provide a PR for such documentation if this really the intended behavior.

**Comment by hakanernam** [comment]
@ctoestreich I just created without the profile

```
grails create-plugin bar
```

And it still shows "web-plugin" as app profile just as the foo plugin created  above with the profile. So It seems think web-profile still is the default. Some clarification would be appreciated.

**Comment by ctoestreich** [comment]
What we found for plugins is that the asset pipeline gradle stuff has to be removed.  You can still apply the plugin proper, but the asset pipeline gradle needs to be added to the consuming parent application ONLY.  Here is some of our build.gradle from a web plugin that works well.

``` gradle
buildscript {
	ext {
		grailsVersion = project.grailsVersion
	}
	repositories {
		maven {
			url "$artifactory_contexturl/$artifactory_repo"
			credentials {
				username artifactory_user
				password artifactory_password
			}
		}
	}
	dependencies {
		classpath "org.grails:grails-gradle-plugin:$grailsVersion"
		classpath(group: 'com.bsb.gradle.plugin', name: 'bsb-build-utils', version: '3.0')
		classpath 'com.moowork.gradle:gradle-node-plugin:0.11'
	}
}

plugins {
	id "io.spring.dependency-management" version "0.5.2.RELEASE"
	id "com.jfrog.artifactory" version "3.1.1"
}

version gitutils.version
group "com.foo"

apply plugin: 'maven-publish'
apply plugin: 'spring-boot'
apply plugin: 'idea'
apply plugin: 'java'
apply plugin: 'war'
apply plugin: "org.grails.grails-plugin"
apply plugin: "org.grails.grails-gsp"

...

dependencies {

     ...
     runtime "org.grails.plugins:asset-pipeline"

}

```

**Comment by ctoestreich** [comment]
Add ` runtime "org.grails.plugins:asset-pipeline"` to your dependencies and see if you get assets processed.

**Comment by ctoestreich** [comment]
@hakanernam We had to go through a learning process to get this working on the 3.x branch as it is different than before and how plugins inherit plugins and how they are used is different than before.  I had a discussion with @davydotcom about this here.  https://github.com/bertramdev/asset-pipeline/issues/60

It looks like the create-plugin using web-plugin doesn't add the asset pipeline gradle or asset pipeline plugin.  At least the plugin is required to run the application inline for sure.

**Comment by hakanernam** [comment]
This seems to be fixed with 3.1.0.
